### PR TITLE
Add whitelist rule for doc

### PR DESCRIPTION
**Pull request by VincentLanglet**
Need https://github.com/sonata-project/dev-kit/pull/792 first

### CODE REVIEW

**Comment by jordisala1991** [comment]
We need to look at the dev kit prs after this merge, some fixes will be needed with those rules.

**Comment by VincentLanglet** [comment]
> We need to look at the dev kit prs after this merge, some fixes will be needed with those rules.

We have time. The PR https://github.com/sonata-project/dev-kit/pull/792 won't be merged before the end of august if I understand correctly ^^

**Comment by jordisala1991** [comment]
There are 2 things, do we really need the other PR first? And if yes, maybe @greg0ire can help moving it forward? we cant wait one month because currently we cant merge PRs on AdminBundle. If this cannot be unblocked, the other option is to revert the adminBundle PR in order to not having failing builds

**Comment by greg0ire** [comment]
> currently we cant merge PRs on AdminBundle.

Sorry I didn't follow everything… this sounds bad, what is it about? :sweat_smile:

**Comment by VincentLanglet** [comment]
> > currently we cant merge PRs on AdminBundle.
> 
> Sorry I didn't follow everything… this sounds bad, what is it about? 😅

`3.x` was added in the doc of SonataAdmin thanks to a whitelist rule.
But dev-kit remove the whitelist rule so now the tests are failing.

This PR re-add the whitelist, but need another PR to be merged before, because the variable is not currently passed.
OskarStar wanted to test my other PR but he's not available before the 23 august.

There is multiple solution
- Updating this PR to pass the variable branch_name to the twig, and fix the conflict on the other branch.
- Merging the other PR now
- Waiting one month (after all, since this PR is pretty similar to the other one, there is no reason for waiting one month for one, but not for the other).

As a quick fix on SonataAdmin, I made https://github.com/sonata-project/SonataAdminBundle/pull/6235

File: templates/project/docs/.doctor-rst.yaml.twig [hunk 1]
```diff
@@ -27,10 +27,12 @@ rules:
     no_bash_prompt: ~
     no_composer_phar: ~
     no_inheritdoc: ~
+    yaml_instead_of_yml_suffix: ~
+    no_app_bundle: ~
+    no_config_yaml: ~
+    kernel_instead_of_app_kernel: ~
+    no_app_console: ~
```
Comment (OskarStark) [inline templates/project/docs/.doctor-rst.yaml.twig:34]: This would bring up a lot of work 🤔

Comment (VincentLanglet) [reply templates/project/docs/.doctor-rst.yaml.twig:34]: There was `# can be enabled when no Symfony 3.4 support anymore` so I thought I could add this without issues.

Do you have any example ?

Comment (OskarStark) [reply templates/project/docs/.doctor-rst.yaml.twig:34]: Lets check it using a Proof PR:
https://github.com/sonata-project/SonataAdminBundle/pull/6357

Comment (OskarStark) [reply templates/project/docs/.doctor-rst.yaml.twig:34]: Wow it looks ok 🤔 

https://github.com/sonata-project/SonataAdminBundle/pull/6357/checks?check_run_id=1050311053

I will try another bundle

Comment (OskarStark) [reply templates/project/docs/.doctor-rst.yaml.twig:34]: Looks good:
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1106/checks?check_run_id=1050324214

https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1106

Lets merge it

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
As the `DOCtor-RST` build is not required, we should check all DevKit PRs for errors.

**Pull request merged**